### ISSUE TITLE
lint: underef gives false positives for multiple levels of indirection

### ISSUE THREAD

**Issue by Quasilyte** [PieselBois/kfulint#68]
For types like `**T` and `***T` underef suggest to remove dereferencing even though it will lead to invalid code. It should take types into account.

Examples:
```go
	type point struct{ x, y int }
	pt := point{}
	pt1 := &pt
	pt2 := &pt1
	pt3 := &pt2
	pt4 := &pt3
	
	_ = (*pt2).x
	_ = (**pt3).x
	_ = (***pt4).x
```

There can be more cases where it gives false positives, but this one can be resolved in isolation.
